### PR TITLE
[navigation] Refactoring and fixing bug in navigation

### PR DESCRIPTION
**Pull request by didierofrivia**
**What this PR does / why we need it**:

This PR fixes the bug `TypeError: Uncaught TypeError: window.hideAllToggleable is not a function`
This was raised because the event assignation in the header partial sometimes was executed before the function assignation in window

**Which issue(s) this PR fixes** 

Fixes https://issues.jboss.org/browse/THREESCALE-1674

### CODE REVIEW

File: app/javascript/packs/navigation.js [hunk 1]
```diff
@@ -2,7 +2,46 @@ import { toggleNavigation, hideAllToggleable } from '../src/Navigation/utils/tog
 import { ContextSelectorWrapper } from '../src/Navigation/components/ContextSelector'
 
 document.addEventListener('DOMContentLoaded', () => {
-  window.toggleNavigation = toggleNavigation
-  window.hideAllToggleable = hideAllToggleable
-  window.ContextSelector = ContextSelectorWrapper
+  const apiSelector = 'api_selector'
+  const apiSelectorNode = document.getElementById(apiSelector)
+  const apiSelectorData = JSON.parse(apiSelectorNode.getAttribute('data'))
```
Comment (hallelujah) [inline app/javascript/packs/navigation.js]: it is only `data`? I thought HTML 5 data attribute should be prefixed by `data-`

Comment (didierofrivia) [reply app/javascript/packs/navigation.js]: It's passed from the view as single data attr: https://github.com/3scale/porta/pull/387/files#diff-c831c6b813f4dcc54721733f25630f1cR1

Comment (hallelujah) [reply app/javascript/packs/navigation.js]: That is not the point. It should be something like `data-something`

File: app/views/shared/_api_selector.html.slim [hunk 2]
```diff
@@ -1,11 +1 @@
...
-javascript:
-  document.addEventListener('DOMContentLoaded', function () {
-    window.ContextSelector({
-      currentApi: #{json @service},
-      apis: #{json api_selector_services},
-      activeMenu: #{json active_menu},
-      audienceLink: #{json audience_link}
-    }, 'api_selector')
-  })
+#api_selector data={currentApi: @service, apis: api_selector_services, activeMenu: active_menu, audienceLink: audience_link}.to_json
```
Comment (hallelujah) [inline app/views/shared/_api_selector.html.slim]: I think you should escape the json

Comment (didierofrivia) [reply app/views/shared/_api_selector.html.slim]: what's the difference with `.to_json`?

Comment (didierofrivia) [reply app/views/shared/_api_selector.html.slim]: I mean, it doesn't also escape js in the process?

Comment (didierofrivia) [reply app/views/shared/_api_selector.html.slim]: also: https://stackoverflow.com/questions/2715994/rails-escape-javascript-creates-invalid-json-by-escaping-single-quotes

**Pull request merged**